api/v1/issues/IssueHandler.inc.php

```diff
@@ -152,47 +157,51 @@ public function getMany($slimRequest, $response, $args)
                     } elseif (!is_array($val)) {
                         $val = [$val];
                     }
-                    $params[$param] = array_map('intval', $val);
+                    $values = array_map('intval', $val);
+                    switch ($param) {
+                        case 'volumes':
+                            $collector->filterByVolumes($values);
+                            break;
+                        case 'numbers':
+                            $collector->filterByNumbers($values);
+                            break;
+                        case 'years':
+                            $collector->filterByYears($values);
+                            break;
+                    }
+
                     break;
 
                 case 'isPublished':
-                    $params[$param] = $val ? true : false;
+                    $isPublishedCheck = (bool) $val;
+                    $collector->filterByPublished((bool) $val);
                     break;
 
                 case 'searchPhrase':
-                    $params[$param] = $val;
+                    $collector->searchPhrase($val);
                     break;
             }
         }
 
-        $params['contextId'] = $context->getId();
+        $collector->filterByContext($context->getId());
 
-        \HookRegistry::call('API::issues::params', [&$params, $slimRequest]);
+        \HookRegistry::call('API::issues::params', [&$collector, $slimRequest]);
```

Comment:
(Since you're touching on this -- there should be  `use PKP\plugins\HookRegistry;` at the head, and the `\` here should be removed)